src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFiltering.cpp

```diff
@@ -314,7 +314,20 @@ namespace OpenMS
           blacklist_[spectrum - 1][peak_index].black_exception_charge = pattern.getCharge();
           blacklist_[spectrum - 1][peak_index].black_exception_mz_position = mz_position;
         }
-
+        
+        // blacklist peaks in spectrum before previous one
+        if (peak_index != -1 && spectrum > 1)
+        {
+          int peak_index_2 = registry_[spectrum - 1][peak_index].index_in_previous_spectrum;
```

Comment:
It's an internal data structure never exposed. In this peak registry one can look up the index of the corresponding peak in the neighbouring spectra.